sharktank/tests/models/llama4/llama4_test_iree_eager.py

```diff
@@ -0,0 +1,39 @@
+from sharktank.models.llama4 import testing
+from sharktank.types import *
+from sharktank.utils.evaluate import *
+
+from sharktank.models.llama.testing import make_random_llama_theta
+from sharktank.utils.testing import TempDirTestBase
+from sharktank.utils.testing import is_mi300x, IreeVsEagerLLMTester, TempDirTestBase
+import torch
+import logging
+
+import pytest
+
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger(__name__)
+
+iree_compile_flags = []  # TODO; fill this if we need any flag
+
+
+@pytest.mark.usefixtures("get_iree_flags", "device")
+@is_mi300x
+class TestLlama4IreeEager(TempDirTestBase):
+    def testUnshardedToySizedModelIREEVsEager(self):
+        seed = 1234
+        random.seed(seed)
+        torch.manual_seed(seed)
+        config = testing.make_toy_model_config(dtype=torch.float16)
```

Comment:
It would be nice to have a f32 test as well with more stringent tolerances.
You could parametrize the test on the dtype, rtol and atol.